Added ./bin/elixir --info option #3947

Troush opened the pull request:
With erlang version, elixir version, elixir revision short hash.
Also added elixir revision short hash to System.build_info().
```
$ ./bin/elixir --version        
Erlang/OTP 18 [erts-7.1] [source] [64-bit] [smp:4:4] [async-threads:10] [hipe] [kernel-poll:false]

Elixir 1.2.0-dev (6b14048)
```
Resolves #3947

josevalim commented:
Excellent! We don't need to add a new flag though. We can show this information with `elixir -v` directly. :)

josevalim commented on lib/elixir/lib/system.ex:
```diff
@@ -42,6 +42,20 @@ defmodule System do
     end
   end
 
+  defmacrop get_revision do
```
This is good. Let's drop the `:tag` from `build_info` and go only with `:revision`. :)

josevalim commented:
I have added some comments, can you please check them out? Thank!

Troush commented:
@josevalim Done all like you said. Deleted tags code and fixed tests.
First --info version was based on comments of #3947 

Local tests are fine, but travis shows some unexpected errors.

josevalim commented on lib/elixir/lib/kernel/cli.ex:
```diff
@@ -146,7 +146,8 @@ defmodule Kernel.CLI do
 
   defp parse_shared([opt|_t], _config) when opt in ["-v", "--version"] do
-    IO.puts "Elixir #{System.version}"
+    IO.puts "Erlang/OTP #{:erlang.system_info(:otp_release)}"
```
Let's do: `IO.puts :erlang.system_info(:system_version)` since it shows more information regarding the environment.

josevalim commented:
One last comment and please don't forget to amend the commit message. Thank you! :)

lexmag commented on lib/elixir/lib/kernel/cli.ex:
```diff
@@ -146,7 +146,8 @@ defmodule Kernel.CLI do
   defp parse_shared([opt|_t], _config) when opt in ["-v", "--version"] do
-    IO.puts "Elixir #{System.version}"
+    IO.puts "Erlang/OTP #{:erlang.system_info(:otp_release)}"
+    IO.puts "Elixir #{System.version} (#{System.build_info()[:revision]})"
```
Little nitpicking, I think `System.build_info.revision` might be easier to read inside many different parens.

josevalim commented:
:heart: :green_heart: :blue_heart: :yellow_heart: :purple_heart:

eksperimental commented:
Wouldn't it by better to have the elixir information in the first line and the Erlang/OTP (since it's additional info) in the third line?

Also if I'm only interested in the elixir version I can easily do `elixir --version | head`
